src/main/java/io/confluent/connect/jdbc/DataConverter.java

```diff
@@ -169,7 +169,27 @@ private static void addFieldSchema(ResultSetMetaData metadata, int col,
         break;
       }
 
-      case Types.NUMERIC:
+      case Types.NUMERIC: {
+        int precision = metadata.getPrecision(col);
+        if (metadata.getScale(col) == 0 && precision < 19) { // integer
```

Comment:
So I can review this properly and validate the choice of exponents, which behavior are we going with to control the range: allowing all valid values for a given integer width but exception/clamping when a value exceeds `<IntType>.MAX_VALUE` or only converting up to a certain range (i.e. disallowing some valid values for `<IntType>` to make sure we can always convert the value we receive to the target type)?